internal/action/generate_dockerfile.go

```diff
@@ -0,0 +1,63 @@
+package action
+
+import (
+	"fmt"
+	"io"
+	"text/template"
+
+	"github.com/operator-framework/operator-registry/pkg/containertools"
+)
+
+type GenerateDockerfile struct {
+	BaseImage   string
+	FromDir     string
+	ExtraLabels map[string]string
+	Writer      io.Writer
+}
+
+func (i GenerateDockerfile) Run() error {
+	if err := i.validate(); err != nil {
+		return err
+	}
+
+	t, err := template.New("dockerfile").Parse(dockerfileTmpl)
+	if err != nil {
+		// The template is hardcoded in the binary, so if
+		// there is a parse error, it was a programmer error.
+		panic(err)
+	}
+	return t.Execute(i.Writer, i)
+}
+
+func (i GenerateDockerfile) validate() error {
+	if i.BaseImage == "" {
+		return fmt.Errorf("base image is unset")
+	}
+	if i.FromDir == "" {
+		return fmt.Errorf("index path is unset")
+	}
+	return nil
+}
+
+const dockerfileTmpl = `# The base image is expected to contain
+# /bin/opm (with a serve subcommand) and /bin/grpc_health_probe
+FROM {{.BaseImage}}
+
+# Configure the entrypoint and command
+ENTRYPOINT ["/bin/opm"]
+CMD ["serve", "/configs"]
+
+# Copy declarative config root into image at /configs
+ADD {{.FromDir}} /configs
+
+# Set DC-specific label for the location of the DC root directory
+# in the image
+LABEL ` + containertools.ConfigsLocationLabel + `=/configs
+{{- if .ExtraLabels }}
+
+# Set other custom labels
+{{- range $key, $value := .ExtraLabels }}
+LABEL "{{ $key }}"="{{ $value }}"
+{{- end }}
+{{- end }}
```

Comment:
Just wanted to clarify the behavior here in case there are multiple key/value pairs that are referenced in the `--extra-labels` flag - if I specify duplicate keys but differing values, it looks like the last value in the chain will always be overriden (e.g. `--extra-labels test=test1,test=test2` would produce a `LABEL test=test2` entry), which seems like the correct behavior. I wasn't sure whether that kind of edge case would belong in the validation logic though.

Reply:
I'll make a note in the help text that the last label wins if there are duplicates.